### PR TITLE
feat(lsp): quick fix actions to ignore lint errors

### LINKED ISSUE #10122: lsp: lint diagnostics should have a quick fix to add ignore comment
Lint diagnostics in the LSP should produce a code action quick fix that inserts a comment to ignore the lint rule.

For example, the following:

```ts
declare var a: any;
```

Which generates the diagnostic:

```
`any` type is not allowed
deno-lint(no-explicit-any)
```

Should have a quick fix which would transform the code to:

```ts
// deno-lint-ignore no-explicit-any
declare var a: any;
```

### PR DESCRIPTION
**Pull request by satyarohith**
Introduces quick-fix actions to ignore lint errors.

https://user-images.githubusercontent.com/29819102/122170586-5a4abe80-ce9c-11eb-95b7-a1c939ffdd70.mp4

Closes #10122

### CODE REVIEW

File: cli/lsp/analysis.rs [hunk 1]
```diff
@@ -479,6 +479,91 @@ impl CodeActionCollection {
...
+            line: diagnostic.range.start.line,
+            character: 0,
+          },
+          end: lsp::Position {
+            line: diagnostic.range.start.line,
+            character: 0,
+          },
+        },
+      }],
+    );
```
Comment (magurotuna) [inline cli/lsp/analysis.rs:513]: Why not `changes.insert(specifier.clone(), vec![...])` ?

File: cli/lsp/analysis.rs [hunk 2]
```diff
@@ -479,6 +479,91 @@ impl CodeActionCollection {
...
+            line: 0,
+            character: 0,
+          },
+          end: lsp::Position {
+            line: 0,
+            character: 0,
+          },
+        },
+      }],
+    );
```
Comment (magurotuna) [inline cli/lsp/analysis.rs:547]: ditto

File: cli/lsp/analysis.rs [hunk 3]
```diff
@@ -479,6 +479,91 @@ impl CodeActionCollection {
...
+    let mut changes = HashMap::new();
+    HashMap::insert(
+      &mut changes,
+      specifier.clone(),
+      vec![lsp::TextEdit {
+        new_text: format!("// deno-lint-ignore {}\n", code),
+        range: lsp::Range {
+          start: lsp::Position {
+            line: diagnostic.range.start.line,
+            character: 0,
```
Comment (dsherret) [inline cli/lsp/analysis.rs:505]: There is a way to get the indentation at the current position via tsserver ("indentaiton" command), but I don't think that would be helpful here though. We may just want to get the index of the first non-whitespace character on the current line and that will be good enough for 99% of cases.

File: cli/lsp/analysis.rs (hunk 1 above)
Comment (satyarohith) [reply cli/lsp/analysis.rs:513]: Not sure why I used them. 😄 

Thanks!

File: cli/lsp/analysis.rs (hunk 3 above)
Comment (satyarohith) [reply cli/lsp/analysis.rs:505]: Ah, this makes sense. I'll see how I can get the index of the first non-whitespace character of the start line.

Thanks for the suggestion! :)

File: cli/lsp/analysis.rs [hunk 4]
```diff
@@ -665,6 +666,102 @@ impl CodeActionCollection {
...
+        .flatten()
+    } else {
+      None
+    };
+
+    let mut changes = HashMap::new();
+    changes.insert(
+      specifier.clone(),
+      vec![lsp::TextEdit {
+        new_text: append_whitespace(
```
Comment (sarcat) [inline cli/lsp/analysis.rs:697]:
```suggestion
        new_text: prepend_whitespace(
```

**Review by kitsonk — CHANGES_REQUESTED** [review]
The only change that I think we need to make is to add a `CodeActionKind` of `DenoLint` since we differ the sources of the lint information as well, so I think we should maintain that separation here.

Otherwise really good, thank you!

**Review by kitsonk — APPROVED** [review]
LGTM

**Pull request merged**